src/Logger/FileLogger.php

```diff
@@ -101,8 +101,17 @@ public function __construct(
 
     public function log(): void
     {
+        $content = implode(PHP_EOL, $this->getLogLines());
+
+        // If the output should be written to a stream then just write it directly
+        if (0 === strpos($this->logFilePath, 'php://')) {
```

Comment:
I would restrict it to `stderr` and `stdout` only, because it doesn't make sense to use e.g. `php://memory` and [other available wrappers](https://www.php.net/manual/en/wrappers.php.php)

I would also suggest to unit test this restriction. We can not throw an exception here, but log with `error` as it's done on the [116 line below](https://github.com/infection/infection/pull/821/files#diff-db63c624738b8c5979eeb77e9fc84314R116)

Reply:
I've re-factored that now